cmake/external/eigen.cmake

```diff
@@ -29,8 +29,10 @@ add_dependencies(eigen3 extern_eigen3)
 
 LIST(APPEND external_project_dependencies eigen3)
 
-IF(NOT WITH_C_API AND WITH_FLUID)
-    INSTALL(FILES ${EIGEN_INCLUDE_DIR}/Eigen/Core DESTINATION third_party/eigen3/Eigen)
-    INSTALL(DIRECTORY ${EIGEN_INCLUDE_DIR}/Eigen/src DESTINATION third_party/eigen3/Eigen)
-    INSTALL(DIRECTORY ${EIGEN_INCLUDE_DIR}/unsupported/Eigen DESTINATION third_party/eigen3/unsupported)
-ENDIF()
+set(lib_dir "${CMAKE_INSTALL_PREFIX}/third_party/eigen3")
+add_custom_target(eigen3_lib
+    COMMAND mkdir -p "${lib_dir}/Eigen" "${lib_dir}/unsupported"
+    COMMAND cp "${EIGEN_INCLUDE_DIR}/Eigen/Core" "${lib_dir}/Eigen"
```

Comment:
It is better to use `${CMAKE_COMMAND} -E copy_directory`, like [here](https://github.com/PaddlePaddle/Paddle/blob/develop/cmake/external/openblas.cmake#L95)